six/modules/c++/six.sidd/source/DerivedXMLParser200.cpp

```diff
@@ -543,7 +543,17 @@ void DerivedXMLParser200::parseLookupTableFromXML(
 void DerivedXMLParser200::parseBandEqualizationFromXML(const XMLElem bandElem,
                                                        BandEqualization& band) const
 {
-    parseEnum(getFirstAndOnly(bandElem, "Algorithm"), band.algorithm);
+    std::string bandAlgo;
+    parseString(getFirstAndOnly(bandElem, "Algorithm"), bandAlgo);
+
+    if (bandAlgo == "1DLUT")
+    {
+        band.algorithm = BandEqualizationAlgorithm("LUT 1D");
+    }
+    else
+    {
+        band.algorithm = BandEqualizationAlgorithm(bandAlgo);
+    }
```

Comment:
this is nasty. is this because of the change in SIDD spec from `LUT 1D` -> `1DLUT`? should this if be added to `BandEqualizationAlgorithm` instead?

Reply:
So the script that generates the enums from a text file requires that the enum be the same as the string. And I'm like 99% sure c++ doesn't allow you to make an enum starting with a number.
I could just go in and manually change the Enums.h file, but if someone re-generates it again the change will be overwritten. 

The script lives somewhere in Coda so I can maybe add functionality to that to optionally specify a different string for each enum but that seemed like a not super essential update to me atm.

Also, sucks that the enum the schema wants is 1DLUT instead of LUT1D ... hopefully we can just change that for the next version.

So yeah open to second opinion.

Reply:
that is fair. makes sense to me. 

i would say this code block is better than changing schema and/or coda.